brian2/core/network.py

```diff
@@ -461,6 +464,32 @@ def add(self, *objs):
                         raise TypeError("Can only add objects of type BrianObject, "
                                         "or containers of such objects to Network")
 
+    def get_all_objects(self):
+
+        all_objects = []
+
+        def add_child_objects(*objs):
+            for obj in objs:
+                if isinstance(obj, BrianObject):
+                    if obj not in all_objects:  # Don't include objects twice
+                        all_objects.append(obj)
+                    add_child_objects(obj.contained_objects)
+                else:
+                    # allow adding values from dictionaries
```

Comment:
This is unecessary here: we want users to be able to add objects from dictionaries when they call `Network(...)` or `Network.add(...)`, but here the function works on an internal attribute, `Network.objects`, which is always a list.